docs/writing-rules/rule-syntax.md

```diff
@@ -81,6 +81,17 @@ The `patterns` operator performs a logical AND operation on one or more child pa
 
 <iframe src="https://semgrep.dev/embed/editor?snippet=Q83q" border="0" frameBorder="0" width="100%" height="432"></iframe>
 
+### Evaluation strategy
+
+Note that the order in which the child patterns are declared in a `patterns` operator has no effect on the final result. A `patterns` operator is always evaluated in the same way:
```

Comment:
@IagoAbal expanding on your sentence, let me know if this is true, please. Does it mean that regardless of how a user defines these patterns in a rule, the order of patterns declared in a rule is arbitrary (for example a rule where `pattern-not` is declared before `patterns`)? Semgrep always evaluates `patterns` in the order you mentioned in these steps, not as the user defines them. Is this correct?

```suggestion
This section documents how Semgrep evaluates rules. This is similar to a funnel where some `patterns` take precedent over other `patterns` when Semgrep is processing the rule. Note that the order in which the child patterns are declared in a `patterns` operator of a rule has no effect on the final result. A `patterns` operator is always evaluated in the same way:
```

Reply:
I feel you're merging "patterns" (as in any pattern operator) and `patterns` (which is a very specific AND operator).

Reply:
The section doesn't really describe how we evaluate any arbitrary rule, but only how we evaluate the `patterns` AND operator.

Reply:
Ah, get it now, thanks. It was not heading level 4 (`####`) which is confusing.